webapp/src/components/Metrics/PerformanceAnalysisTable.test.tsx

```diff
@@ -0,0 +1,230 @@
+      "Extreme Length",
+      "Partial Syntax",
+      "Dissimilar",
+      "Almost Correct",
+      "Behavioral Testing",
+      "Pipeline Comparison",
+      "Uncertain",
+    ];
+    expect(filterList).toHaveLength(expectedList.length);
+    filterList.forEach((item, index) => {
+      expect(item).toHaveTextContent(expectedList[index]);
+    });
+
+    expect(screen.getByRole("option", { name: "Smart Tags" })).toHaveAttribute(
+      "aria-selected",
+      "false"
+    );
+  });
+});
+
+describe("PerformanceAnalysisComparisonTableWithMultiPipeline", () => {
+  const handlers = [
+    getConfigMultipipelineAPIResponse,
+    getCustomMetricInfoAPIResponse,
+    getMetricsPerFilterAPIResponse,
+  ];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  it("should display pipeline select component if the config has more than one pipeline", async () => {
+    renderPerformanceAnalysisTable({ train: true, eval: true }, "eval");
+    await screen.findByText("Compare Baseline (Pipeline_0) with:");
+  });
+
+  it("should display list of pipelines", async () => {
+    renderPerformanceAnalysisTable({ train: true, eval: true }, "eval");
+
+    await waitFor(() => {
+      fireEvent.mouseDown(screen.getByRole("button", { name: "No pipeline" }));
+      const pipelineList = screen.getAllByRole("option");
+      const expectedPipelineList = [
+        "No pipeline",
+        "Pipeline_0",
+        "Pipeline_1",
+        "Pipeline_2",
+        "Pipeline_3",
+      ];
+      expect(pipelineList).toHaveLength(expectedPipelineList.length);
+      pipelineList.forEach((item, index) => {
+        expect(item).toHaveTextContent(expectedPipelineList[index]);
+      });
+      expect(screen.getByRole("option", { name: "Pipeline_0" })).toHaveClass(
+        "Mui-disabled"
+      );
+    });
+  });
+
+  it("should display the expected columns", async () => {
+    renderPerformanceAnalysisTable({ train: true, eval: true }, "eval");
+    fireEvent.mouseDown(
+      await screen.findByRole("button", { name: "No pipeline" })
+    );
+    fireEvent.click(await screen.findByRole("option", { name: "Pipeline_1" }));
+    await waitFor(() => {
+      const expectedColumnHeaders = [
+        "filterValue",
+        "Total",
+        "Pipeline_0Correct & Predicted",
+        "Pipeline_1",
+        "Delta",
+        "Pipeline_0Correct & Rejected",
+        "Pipeline_1",
+        "Delta",
+        "Pipeline_0Incorrect & Rejected",
+        "Pipeline_1",
+        "Delta",
+        "Pipeline_0Incorrect & Predicted",
+        "Pipeline_1",
+        "Delta",
+        "Accuracy",
+        "Pipeline_1",
+        "Delta",
+        "Precision",
+        "Pipeline_1",
+        "Delta",
+        "Recall",
+        "Pipeline_1",
+        "Delta",
+        "F1",
+        "Pipeline_1",
+        "Delta",
+        "ECE",
+        "Pipeline_1",
+        "Delta",
```

Comment:
I notice `Pipeline_0` is verified for 7 columns, but missing for 5.
```suggestion
        "Pipeline_0Correct & Predicted",
        "Pipeline_1",
        "Delta",
        "Pipeline_0Correct & Rejected",
        "Pipeline_1",
        "Delta",
        "Pipeline_0Incorrect & Rejected",
        "Pipeline_1",
        "Delta",
        "Pipeline_0Incorrect & Predicted",
        "Pipeline_1",
        "Delta",
        "Pipeline_0Accuracy",
        "Pipeline_1",
        "Delta",
        "Pipeline_0Precision",
        "Pipeline_1",
        "Delta",
        "Pipeline_0Recall",
        "Pipeline_1",
        "Delta",
        "Pipeline_0F1",
        "Pipeline_1",
        "Delta",
        "Pipeline_0ECE",
        "Pipeline_1",
        "Delta",
```
Alternatively, we can simply with something like that:
```suggestion
        ...[
          "Correct & Predicted",
          "Correct & Rejected",
          "Incorrect & Rejected",
          "Incorrect & Predicted",
          "Accuracy",
          "Precision",
          "Recall",
          "F1",
          "ECE",
        ].flatMap((header) => ["Pipeline_0" + header, "Pipeline_1", "Delta"]),
```

Reply:
Ohh Wow! Yes. I like this simplified way. A good one to learn. Thanks